src/handlers/ripgrep_json.rs

```diff
@@ -15,7 +15,16 @@ pub fn parse_line(line: &str) -> Option<grep::GrepLine> {
             // A real line of rg --json output, i.e. either of type "match" or
             // "context".
             let mut code = ripgrep_line.data.lines.text;
-            if code.ends_with('\n') {
+            // ripgrep --json emits a trailing newline for each match, but
+            // delta's code highlighting does not expect a trailing newline.
+            // Therefore, remove the trailing newline if it is not part of the
+            // match.
+            let newline_match_end = ripgrep_line
+                .data
+                .submatches
+                .iter()
+                .any(|m| code.ends_with(&m._match.text) && m._match.text.ends_with('\n'));
```

Comment:
He might have multiple matches and it wouldn't always be the last match or am i missing something ?
```
[{"match":{"text":"\n\n\n"},"start":60,"end":63}}, {"match":{"text":"config"},"start":50,"end":55}}
```
This is not an actual example

Reply:
OK sorry I'll try to find time to think through this properly! Your PR looks great, the only reason I've been slow is because I was slightly confused about exactly what the condition should be for us stripping newline.

Reply:
No worries ! in any case keep me updated ! thanks